### PR TITLE
add a git hook to linting 

### PR DESCRIPTION
**Pull request by goat023**
Add a git hook to lint the codes

closes https://github.com/apache/skywalking/issues/6034

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,7 +14,10 @@
     "format": "prettier --write \"src/**/*.ts\"",
     "clean": "(rm -rf src/proto || true) && (rm -rf src/proto || true) && (rm -rf lib || true)",
     "package-src": "tar -zcvf skywalking-nodejs-src-$npm_package_version.tgz --exclude bin --exclude .git --exclude .idea --exclude .DS_Store --exclude .github --exclude node_modules --exclude skywalking-nodejs-src-$npm_package_version.tgz .",
-    "release-src": "npm run prepare && npm run package-src && gpg --batch --yes --armor --detach-sig skywalking-nodejs-src-$npm_package_version.tgz && shasum -a 512 skywalking-nodejs-src-$npm_package_version.tgz > skywalking-nodejs-src-$npm_package_version.tgz.sha512"
+    "release-src": "npm run prepare && npm run package-src && gpg --batch --yes --armor --detach-sig skywalking-nodejs-src-$npm_package_version.tgz && shasum -a 512 skywalking-nodejs-src-$npm_package_version.tgz > skywalking-nodejs-src-$npm_package_version.tgz.sha512",
+    "lint:js": "eslint . --ext .js",
```
Comment (kezhenxu94) [inline package.json:18]: We should lint `*.ts`, not `*.js`

Comment (goat023) [reply package.json:18]: oh sorry, will change it.

File: package.json [hunk 2]
```diff
@@ -14,7 +14,10 @@
     "format": "prettier --write \"src/**/*.ts\"",
     "clean": "(rm -rf src/proto || true) && (rm -rf src/proto || true) && (rm -rf lib || true)",
     "package-src": "tar -zcvf skywalking-nodejs-src-$npm_package_version.tgz --exclude bin --exclude .git --exclude .idea --exclude .DS_Store --exclude .github --exclude node_modules --exclude skywalking-nodejs-src-$npm_package_version.tgz .",
-    "release-src": "npm run prepare && npm run package-src && gpg --batch --yes --armor --detach-sig skywalking-nodejs-src-$npm_package_version.tgz && shasum -a 512 skywalking-nodejs-src-$npm_package_version.tgz > skywalking-nodejs-src-$npm_package_version.tgz.sha512"
+    "release-src": "npm run prepare && npm run package-src && gpg --batch --yes --armor --detach-sig skywalking-nodejs-src-$npm_package_version.tgz && shasum -a 512 skywalking-nodejs-src-$npm_package_version.tgz > skywalking-nodejs-src-$npm_package_version.tgz.sha512",
+    "lint:ts": "eslint . --ext .ts",
```
Comment (kezhenxu94) [inline package.json:18]: We don't use eslint, take a look at line 12, we use `tslint`, so `npm run lint:ts` will fail always `sh: eslint: command not found`

File: package.json [hunk 3]
```diff
@@ -14,7 +14,10 @@
...
+    "lint:ts:fix": "npm run lint:ts -- --fix",
+    "test:related": "jest --bail --findRelatedTests"
```
Comment (kezhenxu94) [inline package.json:20]: Why do we need this?

File: .huskyrc.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+
+module.exports = {
+    hooks: {
+      "pre-commit": "lint-staged",
+    },
+  };
```
Comment (kezhenxu94) [inline .huskyrc.js:6]: format this please, and add license header to this file

File: .lintstagedrc.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+module.exports = {
+    "src/**/*.ts": ["npm run lint:ts", "npm run test:related"],
+  };
```
Comment (kezhenxu94) [inline .lintstagedrc.js:4]: format this please, and add license header to this file

**Review by kezhenxu94 — COMMENTED** [review]
According to the doc, hasky is not enabled for new developers, so can you please take a look at [this](https://typicode.github.io/husky/#/) and add a postinstall goal to `package.json` so that new developers can install it automatically?

> To automatically have Git hooks enabled after install, edit package.json
>
> // package.json
> 
> ```json
> {
>  "private": true,
>  "scripts": {
>    "postinstall": "husky install"
>  }
> }
> ```

**Comment by wu-sheng** [comment]
Seems no update? Should we close this?

**Comment by kezhenxu94** [comment]
Don’t think the author is still working on this

**Pull request closed without merging**